src/charm.py

```diff
@@ -186,6 +187,14 @@ def _on_certificate_creation_request(self, event: CertificateCreationRequestEven
         )
         logger.info(f"Generated certificate for relation {event.relation_id}")
 
+    def _on_get_ca_cert(self, event):
+        if not self._root_certificate_is_stored:
+            event.set_results({"result": "Root Certificate is not yet generated"})
```

Comment:
Shouldn't we also `fail` in these cases (`event.fail(...)`)?

Reply:
Done.